espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Handlers/ScrollTo.java

```diff
@@ -0,0 +1,55 @@
+package io.appium.espressoserver.lib.Handlers;
+
+import android.support.test.espresso.ViewInteraction;
+
+import java.util.Map;
+
+import fi.iki.elonen.NanoHTTPD;
+import io.appium.espressoserver.lib.Http.Response.AppiumResponse;
+import io.appium.espressoserver.lib.Http.Response.BaseResponse;
+import io.appium.espressoserver.lib.Http.Response.InternalErrorResponse;
+import io.appium.espressoserver.lib.Http.Response.InvalidSessionResponse;
+import io.appium.espressoserver.lib.Model.AppiumStatus;
+import io.appium.espressoserver.lib.Model.Session;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.scrollTo;
+import static android.support.test.espresso.action.ViewActions.
+import static android.support.test.espresso.matcher.ViewMatchers.withText;;
+
+/**
+ * Created by ahmetkocu on 27.05.2017.
+ */
+
+public class ScrollTo implements RequestHandler {
+
+    @Override
+    public BaseResponse handle(NanoHTTPD.IHTTPSession session, Map<String, String> uriParams) {
+
+        // If the SessionID is invalid, return InvalidSessionResponse
+        // TODO: Fix SessionID handling redundancies
+        if (!uriParams.get("sessionId").equals(Session.getGlobalSessionId())) {
+            return new InvalidSessionResponse(uriParams.get("sessionId"));
+        }
+
+        String text = uriParams.get("text").toString();
```

Comment:
I'm about to push a PR that refactors the way handlers work. Instead of using hashmaps, it will use objects that are deserialized from the JSON string so that we can avoid potential errors.